### ISSUE TITLE
Error when setting permissions based on Ecto.Enum attribute

### ISSUE THREAD

**Issue by toberoni** [zachallaun/janus#3]
Hi,

I experienced some problems when setting permissions for a Story schema based on a :status Ecto.Enum field:

```elixir
 #policy.ex
  def build_policy(policy, _actor) do
    policy
    |> allow(Story, :read, where: [status: :public])
  end
```

1. The above works when I fetch a single Story with my custom `authorized_fetch_one` which is very similar to the default `authorized_fetch_by`. I also tried it with other values for :status and can confirm it works as expected.
2. It works without the where: filter in `allow()`
3. It blows up when I use `scope` or `authorized_fetch_all` with the following error message:

```bash
17:48:31.335 [error] GenServer #PID<0.808.0> terminating
** (FunctionClauseError) no function clause matching in Janus.Authorization.Filter.dump/2
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:131: Janus.Authorization.Filter.dump({:parameterized, {Ecto.Enum, %{on_load: %{"created" => :created, "finished" => :finished, "in_review" => :in_review, "next_audio" => :next_audio, "next_illustrations" => :next_illustrations, "public" => :public}, type: :string, on_cast: %{"created" => :created, "finished" => :finished, "in_review" => :in_review, "next_audio" => :next_audio, "next_illustrations" => :next_illustrations, "public" => :public}, embed_as: :self, mappings: [created: "created", next_illustrations: "next_illustrations", next_audio: "next_audio", finished: "finished", in_review: "in_review", public: "public"], on_dump: %{public: "public", finished: "finished", created: "created", in_review: "in_review", next_audio: "next_audio", next_illustrations: "next_illustrations"}}}}, :public)
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:125: Janus.Authorization.Filter.dump!/2
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:119: Janus.Authorization.Filter.dynamic_compare/3
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:101: Janus.Authorization.Filter.apply_filter_clause/2
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:52: anonymous fn/4 in Janus.Authorization.Filter.combine_all/4
    (elixir 1.17.2) lib/enum.ex:2531: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:52: anonymous fn/4 in Janus.Authorization.Filter.combine_all/4
    (elixir 1.17.2) lib/enum.ex:2531: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:52: anonymous fn/4 in Janus.Authorization.Filter.combine_all/4
    (elixir 1.17.2) lib/enum.ex:2531: Enum."-reduce/3-lists^foldl/2-0-"/3
    (ex_janus 0.3.2) lib/janus/authorization/filter.ex:41: Janus.Authorization.Filter.filter_query/4
    (app 0.1.0) lib/app/policy.ex:114: App.Policy.authorized_fetch_all/2
```
I use a Sqlite3 database, not sure if this is relevant. I hope this is enough information to fix the problem, though I'm happy to provide more infos. Thank you very much for this library!

**Comment by zachallaun** [comment]
Thanks for the report, @toberoni!

If you're able to set up a minimal project that I can run to reproduce the error, that would greatly help me in tracking things down.

**Comment by toberoni** [comment]
Hi @zachallaun! 

Here is an example app. The 2 functions in the `Stories` context reproduce the error and also the successful fetch of only 1 record:

https://github.com/toberoni/janus_example

Hope this is helpful!

**Comment by zachallaun** [comment]
Thank you!

This is the result of a change to the internal representation of parameterized types in 3.12 ([described here](https://hexdocs.pm/ecto/changelog.html#potential-incompatibilities)). Should be an easy fix; will get a release out that solves it ASAP.

**Comment by zachallaun** [comment]
When you have a moment, could you please try out the `v0.3` branch and confirm that the included fix works for you?

```elixir
{:ex_janus, github: "zachallaun/ex_janus", ref: "v0.3"}
```

Once confirmed, I'll cut a release!

**Comment by toberoni** [comment]
Hi Zach,

I can confirm this branch fixes the problem. Thank you very much for the fast solution!

**Comment by zachallaun** [comment]
Released as v0.3.3 🙂
